config/crd/bases/postgresql.cnpg.io_clusters.yaml

```diff
@@ -1220,6 +1220,11 @@ spec:
                           (i.e., timestamp, transaction Id, LSN, ...). More info:
                           https://www.postgresql.org/docs/current/runtime-config-wal.html#RUNTIME-CONFIG-WAL-RECOVERY-TARGET'
                         properties:
+                          backupID:
+                            description: 'The backup ID to restore when one of the
+                              following targets is specified: targetName, targetXID
+                              and targetImmediate.'
```

Comment:
Mhhh ... Why not target time too? what if I want to restore using a previous backup?

Reply:
Because it does not make sense to me. Could you please elaborate more with a use case when this could be necessary?

Reply:
Because the closest backup might be corrupted and you want to restore from the previous one. Rare case, but still I would not prevent the choice from the start. It's actually simpler to explain. If you provide the Backup ID we always take that.

Reply:
This was my initial input:

> As a result, it makes sense to add another option called backupId in the recoveryTarget section to be used only when source is specified. By default backupId is empty, meaning that the operator will automatically detect the backup from which to start the recovery in case targetTime is specified, or use the latest available. If non empty, the operator will not perform the automated detection algorithm and always use that backup ID as start of the recovery process - entirely leaving the responsibility to the end user.

See #233